server/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -301,6 +320,9 @@ private void sendResponse(
         final TimeoutState timeoutState,
         final ActionListener<ClusterHealthResponse> listener
     ) {
+        // Creating the ClusterHealthResponse below can be computationally heavy. Ensure this thread is not running on a hot path thread.
+        assert ThreadPool.assertCurrentThreadPool(ThreadPool.Names.MANAGEMENT);
```

Comment:
Seems to fail in [this CI test](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+part-1/40761/consoleFull) at least. Would it be better to fork also this one?